src/jinja2/environment.py

```diff
@@ -1120,7 +1120,13 @@ def make_module_async(self, vars=None, shared=False, locals=None):
         )
 
     @internalcode
-    def _get_default_module(self):
+    def _get_default_module(self, ctx=None):
+        if ctx is not None:
```

Comment:
Is it intentional that the module is not cached as `self._module` in this case? If so, the comment should explain why.

Reply:
Yep was intentional. Added it to the comments 👍